app/backend/src/couchers/models.py

```diff
@@ -1216,3 +1216,18 @@ def ready_for_retry(self):
 
     def __repr__(self):
         return f"BackgroundJob(id={self.id}, job_type={self.job_type}, state={self.state}, next_attempt_after={self.next_attempt_after}, try_count={self.try_count}, failure_info={self.failure_info})"
+
+
+class UserBlocking(Base):
+    """
+    Table of blocked users
+    """
+
+    __tablename__ = "user_blocking"
+
+    id = Column(BigInteger, primary_key=True)
+
+    blocking_user = Column(ForeignKey("users.id"), nullable=False)
+    blocked_user = Column(ForeignKey("users.id"), nullable=False)
```

Comment:
These should be postfixed with `_id`, and you should create relationships based on them

This table also needs a uniqueness constraint each user can block another user at most once.

Reply:
>create relationships based on them

How do you mean?

>This table also needs a uniqueness constraint each user can block another user at most once.

In api.user_blocking(), it checks if the user has already been blocked and doesn't create a new block. That being said, happy to figure out how to add a uniqueness constraint

Reply:
> How do you mean?

Add SQLAlchemy `relationship` objects so you can use them easier in SQLA code, something like this:

https://github.com/Couchers-org/couchers/blob/620a41fd8d2ca07b5456d2f8deaacc0a356ab44d/app/backend/src/couchers/models.py#L450-L451

> In api.user_blocking(), it checks if the user has already been blocked and doesn't create a new block. That being said, happy to figure out how to add a uniqueness constraint

Yes, but it'd be nice to have an additional database check in place so we can guarantee the data is correct. Similar to this:

https://github.com/Couchers-org/couchers/blob/620a41fd8d2ca07b5456d2f8deaacc0a356ab44d/app/backend/src/couchers/models.py#L835

Reply:
I'm having trouble understanding how the relationship works/should work. Specifically, both blocking_user_id and blocked_user_id reference the same model.field and I couldn't figure out how that works.

Reply:
The relationships allow you to do things like

```py3
block = session.query(UserBlock).filter(...).one()

if block.blocking_user.username == "itsi":
  raise Exception("No one can block Itsi")
```

Where sqlalchemy will do the automagic to get the blocking user's username from the db.



> Specifically, both blocking_user_id and blocked_user_id reference the same model.field and I couldn't figure out how that works.

You need to specify a `foreign_keys` value as in the snippet above, so SA knows which row to use